airbyte-integrations/connectors/source-mongodb-internal-poc/src/main/java/io/airbyte/integrations/source/mongodb/internal/MongoDbSource.java

```diff
@@ -94,72 +90,24 @@ public AirbyteCatalog discover(final JsonNode config) {
   public AutoCloseableIterator<AirbyteMessage> read(final JsonNode config,
                                                     final ConfiguredAirbyteCatalog catalog,
                                                     final JsonNode state) {
-    final var databaseName = config.get(DATABASE_CONFIGURATION_KEY).asText();
     final var emittedAt = Instant.now();
     final var stateManager = MongoDbStateManager.createStateManager(state);
-    final MongoClient mongoClient = MongoConnectionUtils.createMongoClient(config);
+    // WARNING: do not close the client here since it needs to be used by the iterator
+    final MongoClient mongoClient = createMongoClient(config);
```

Comment:
Shouldn't the `mongoClient` be closed after intial snapshot is complete? Why do we wait for Debezium to finish as well for to close this. I guess Debezium doesn't use this and constructs its own connection right?